cartoframes/dataset.py

```diff
@@ -105,6 +107,29 @@ def set_geodataframe(self, gdf):
     def get_table_name(self):
         return self._table_name
 
+    def get_dataset_info(self):
+        if self._is_saved_in_carto:
+            if self._dataset_info is None:
+                self._dataset_info = self._get_dataset_info()
+
+            return self._dataset_info
+        else:
+            raise CartoException('Your data is not synchronized with CARTO.'
+                                 'First of all, you should call upload method to save your data in CARTO.')
+
+    def set_dataset_info(self, dataset_info=None, privacy=None, name=None):
+        if self._is_saved_in_carto:
+            if self._dataset_info is None:
```

Comment:
Looks like duplicated code, could we use the public method `get_dataset_info` to avoid these checks?

Reply:
:+1: